developing-fcc-software/FccSoftwareGit.md

```diff
@@ -14,81 +14,148 @@ For a general introduction to git, have a look at these tutorials:
 :::
 
 
-## Improving your git experience
+### Improving your git experience
 
 It may be useful to install [Git integration tools](https://github.com/git/git/tree/master/contrib/completion) for your shell that allow tab-completion of most git commands and also can show you in your prompt on which branch you currently are, what changes you have pending, etc.
 
-## Development workflow
+Other tools with a graphical interface are available, e.g., `git-gui`, `git-cola`, `gitkraken` (which present the main Git commands as buttons, shows differences between commits, etc.) and `tig` (a terminal-based tool that graphically displays the Git commit history).
+
+## How to contribute
+
+This section covers the main commands to contribute to an open project via Pull Requests. See [TLDR section](TL;DR) for a summary.
 
-You will be using (at least) 3 versions of the FCCSW repository:
+### Understanding repositories
 
-1.  The official [FCCSW on github](https://github.com/HEP-FCC/FCCSW)
-2.  Your fork of FCCSW on github (see [github help](https://help.github.com/articles/fork-a-repo/) on what that means)
-3.  Your local repository copy in your work area (e.g. on AFS)
+You will be using (at least) 3 versions of a repository, two on GitHub and a local copy in your machine:
 
-The repositories 1 and 2 are added as remote to the repository 3:
+1.  The official on GitHub, e.g. [FCCSW on github](https://github.com/HEP-FCC/FCCSW)
+2.  Your fork on GitHub (see [github help](https://help.github.com/articles/fork-a-repo/) on what that means)
+3.  Your local repository copy in your work area
 
+The repositories 1 and 2 are added as remote to the repository 3. Let's start by creating a local copy of a dummy repository called `key4hep_repo`, from the `key4hep` organization
+
+```bash
+git clone git@github.com:[YOUR_GITHUB_USER]/key4hep_repo.git
+```
+
+That command links the local copy to the remote repository. The default name of this remote repository is `origin`. Let's add the officialrepository as remote too, but with a different name. The usual convention is to name the official repository `upstream`:
 ```bash
-git clone git@github.com:[YOUR_GITHUB_USER]/FCCSW.git # create a local copy (3) of your fork (2)
 cd FCCSW
-git remote add hep-fcc git@github.com:HEP-FCC/FCCSW.git # add official repo (1) as additional remote
+git remote add upstream https://github.com/key4hep/key4hep_repo.git
 ```
 
-### Keeping your local repository up to date
+### Development workflow
 
--   fetch all changes from the official repository (1)
+This section shows how to work with a non-global package, how to develop a change and upstream it to the official repository.
```

Comment:
What non-global means is not very clear.

Reply:
true! Thanks for spotting it. I will replace the whole sentence by the following, is it better?
This section explains how to use a locally installed package instead of the one provided in the stack, make changes to it, and contribute those changes to the official repository.